### PR TITLE
[13.0][ADD][WIP] purchase_stock_picking_invoice_link

### PR DESCRIPTION
**Pull request by dalonsod**
Adds links between purchase invoices and pickings. Based on #739 (v12) and  #606 (v11)

This is a work in progress (tests not working yet).

### CODE REVIEW

**Comment by dalonsod** [comment]
@pedrobaeza @cubells could you review?

I also need some help making tests: I've tried a lot of changes, but I'm not able to create an invoice linked to the test purchase order, looked for examples across Odoo and OCA, but nothing found. Could you help me? Thanks in advance!

File: purchase_stock_picking_invoice_link/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+{
+    "name": "Purchase Stock Picking Invoice Link",
+    "version": "13.0.1.0.0",
+    "category": "Warehouse Management",
+    "summary": "Adds link between purchases, pickings and invoices",
+    "author": "Tecnativa, " "Odoo Community Association (OCA)",
+    "website": "http://github.com/OCA/stock-logistics-workflow",
+    "license": "AGPL-3",
+    "depends": ["stock_picking_invoice_link", "purchase"],
+    "data": [],
```
Comment (ernestotejeda) [inline purchase_stock_picking_invoice_link/__manifest__.py:13]:
```suggestion
```

File: purchase_stock_picking_invoice_link/tests/test_purchase_stock_picking_invoice_link.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                            "name": product.name,
+                            "product_qty": 1.0,
+                            "date_planned": fields.date.today(),
+                            "product_uom": product.uom_po_id.id,
+                            "price_unit": 15.0,
+                        },
+                    )
+                ],
+            }
+        )
```
Comment (ernestotejeda) [inline purchase_stock_picking_invoice_link/tests/test_purchase_stock_picking_invoice_link.py:33]:
```suggestion
        po_form = Form(cls.env["purchase.order"])
        po_form.partner_id = cls.supplier
        with po_form.order_line.new() as po_line_form:
            po_line_form.product_id = product
            po_line_form.price_unit = 15.0
        cls.po = po_form.save()
```

File: purchase_stock_picking_invoice_link/tests/test_purchase_stock_picking_invoice_link.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            {
+                "partner_id": self.supplier.id,
+                "purchase_id": self.po.id,
+                "company_id": self.po.company_id.id,
+                "type": "in_invoice",
+                "invoice_origin": self.po.name,
+                "ref": self.po.partner_ref,
+            }
+        )
+        self.invoice._onchange_purchase_auto_complete()
```
Comment (ernestotejeda) [inline purchase_stock_picking_invoice_link/tests/test_purchase_stock_picking_invoice_link.py:54]:
```suggestion
        invoice_form = Form(
            self.env["account.move"].with_context(default_type="in_invoice")
        )
        invoice_form.purchase_id = self.po
        self.invoice = invoice_form.save()
```

**Review by ernestotejeda — COMMENTED** [review]
thanks @dalonsod .
I prefer to use the tests.Form class in unit tests to create some objects like, purchase order, sales order, invoice ...
Try those `suggested changes` ;)

File: purchase_stock_picking_invoice_link/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# Copyright 2019 Vicent Cubells <vicent.cubells@tecnativa.com>
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+{
+    "name": "Purchase Stock Picking Invoice Link",
+    "version": "13.0.1.0.0",
+    "category": "Warehouse Management",
+    "summary": "Adds link between purchases, pickings and invoices",
+    "author": "Tecnativa, " "Odoo Community Association (OCA)",
```
Comment (ernestotejeda) [inline purchase_stock_picking_invoice_link/__manifest__.py:9]:
```suggestion
    "author": "Tecnativa, Odoo Community Association (OCA)",
```

**Comment by dalonsod** [comment]
@ernestotejeda thanks a lot for the suggestions 😃 , I was blocked with it, I'll test them. I'll also make other modifications, the code was still incomplete.

**Comment by ernestotejeda** [comment]
Hi @dalonsod , are you going to continue with this PR? If you can't, I can make a PR that supersede this one based on what you've done. Let me now, thanks.

**Comment by dalonsod** [comment]
Hello @ernestotejeda . Finally it's been impossible for me resuming this PR, so I'm glad if you continue with it. Thanks!

**Comment by pedrobaeza** [comment]
Superseded by #883

**Pull request closed without merging**